### DISCUSSION TITLE
Continue training after 1000 epochs

### DISCUSSION THREAD

**Discussion by arthursw** [MIC-DKFZ/nnUNet#1138] [Q&A] [answered]
Hi,

I trained a network for 1000 epochs, then restart the training with `NetworkTrainer.max_num_epochs = 2000` and the `--continue` option.

I got the following progress, with a big bump at epoch 1000:

![image](https://user-images.githubusercontent.com/1814799/185589386-8661fee0-69d7-41ff-a59d-2b53c0f75f0c.png)

I guess this is just due to the learning rate which was almost zero (1.99e-5 at epoch 999) before epoch 1000, and jumped at 5.36e-3 at epoch 1000 when I restarted the training? (from the learning rate function in `nnunet/training/learning_rate/poly_lr.py`)

I should have set `NetworkTrainer.initial_lr` to 1.99e-5 before resuming the training?

**Comment by arthursw** [comment]
Hum, starting from 1.99e-5 does not look good either...

![image](https://user-images.githubusercontent.com/1814799/185593061-e8d7550f-0e0b-4d6a-848f-4644694ab6ae.png)

So I don't know what is advised? To restart the whole training from the beginning with `max_num_epochs==2000`?

**Comment by arthursw** [comment, marked as answer]
from @FabianIsensee https://github.com/MIC-DKFZ/nnUNet/issues/901#issuecomment-1008896695:

> You can of course build workarounds like the one you proposed but my fear is that none of them could be perfect.
To me the easiest solution would be the folowing:
>  - start some training for 1000 epochs, after XX epochs you notice that you need to train longer
>  - create a new nnunet trainer with max_num_epochs equal to the remaining number of epochs you want to train (so if you stop at epoch 500 and want to train for 5000 then 4500 would be what you set)
>  - give this trainer a lower initial learning rate. You could set it to the last learning rate from previous training, but the training may already have progresses too much so that the lr is too low. In that case give it something that is lower than nnU-Net's standard initial lr (may need tuning)
>  - start a new training with your new trainer class, use pretrained model weights from unfinished training as initialization
